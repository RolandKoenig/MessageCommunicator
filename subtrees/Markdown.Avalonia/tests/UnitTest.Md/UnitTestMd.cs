using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using UnitTest.Base;
using UnitTest.Base.Utils;

namespace UnitTest.Md
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTestMd : UnitTestBase
    {
        [Test]
        [RunOnUI]
        public void Transform_givenTest1_generatesExpectedResult()
        {
            var text = Util.LoadText("Test1.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists1_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists1.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists2_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists2.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists3_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists3.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenTables1_generatesExpectedResult()
        {
            var text = Util.LoadText("Tables.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenHorizontalRules_generatesExpectedResult()
        {
            var text = Util.LoadText("HorizontalRules.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLinksInline_generatesExpectedResult()
        {
            var text = Util.LoadText("Links_inline_style.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenTextStyles_generatesExpectedResult()
        {
            var text = Util.LoadText("Text_style.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenImages_generatesExpectedResult()
        {
            var text = Util.LoadText("Images.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenBlockqoute_generatesExpectedResult()
        {
            var text = Util.LoadText("Blockquite.md");
            var markdown = new Markdown.Avalonia.Markdown();

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenMixing_generatesExpectedResult()
        {
            var text = Util.LoadText("Mixing.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenCodes_generatesExpectedResult()
        {
            var text = Util.LoadText("Codes.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenEmoji()
        {
            var text = Util.LoadText("Emoji.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }
    }

}