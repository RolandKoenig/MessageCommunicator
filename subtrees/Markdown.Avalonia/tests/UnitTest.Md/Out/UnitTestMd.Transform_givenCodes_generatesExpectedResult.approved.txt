<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:m="clr-namespace:Markdown.Avalonia.Controls;assembly=Markdown.Avalonia" xmlns:a="https://github.com/avaloniaui/avaloniaedit" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <c:CTextBlock Classes="Heading1">
    <c:CCode>
      <c:CRun Text="```" />
    </c:CCode>
  </c:CTextBlock>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text="```cs&#xA;public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}&#xA;```" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text="public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <Border Classes="Blockquote" xmlns="">
    <Border.Child>
      <StackPanel Classes="Blockquote">
        <Border Classes="CodeBlock">
          <Border.Child>
            <ScrollViewer Classes="CodeBlock">
              <ScrollViewer.Content>
                <TextBlock Text="public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}" Classes="CodeBlock" />
              </ScrollViewer.Content>
            </ScrollViewer>
          </Border.Child>
        </Border>
      </StackPanel>
    </Border.Child>
  </Border>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text="   public void Main(){&#xA;       Console.WriteLine(&quot;`Hello World`&quot;);&#xA;   }" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text="```cs&#xA;public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}&#xA;```" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <Border Classes="CodeBlock">
        <Border.Child>
          <ScrollViewer Classes="CodeBlock">
            <ScrollViewer.Content>
              <TextBlock Text="public void Main(){&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;}" Classes="CodeBlock" />
            </ScrollViewer.Content>
          </ScrollViewer>
        </Border.Child>
      </Border>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <m:Rule BoldLineWidth="2" Classes="Bold" />
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="▪" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="one" />
      </c:CTextBlock>
      <Border Classes="CodeBlock">
        <Border.Child>
          <ScrollViewer Classes="CodeBlock">
            <ScrollViewer.Content>
              <TextBlock Text="public void Main(){&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;}" Classes="CodeBlock" />
            </ScrollViewer.Content>
          </ScrollViewer>
        </Border.Child>
      </Border>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <m:Rule BoldLineWidth="2" Classes="TwoLines" />
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="▪" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="one" />
      </c:CTextBlock>
      <Border Classes="CodeBlock">
        <Border.Child>
          <ScrollViewer Classes="CodeBlock">
            <ScrollViewer.Content>
              <TextBlock Text="public void Main(){&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;}" Classes="CodeBlock" />
            </ScrollViewer.Content>
          </ScrollViewer>
        </Border.Child>
      </Border>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <c:CTextBlock Classes="Heading1">
    <c:CRun Text="Indent" />
  </c:CTextBlock>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text="abcd&#xA;efgh" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <m:Rule BoldLineWidth="2" Classes="Single" />
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text=" abcd&#xA;efgh" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <m:Rule BoldLineWidth="2" Classes="Single" />
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text="abcd&#xA; efgh&#xA;&#xA; ijkl&#xA;&#xA;opqr" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <m:Rule BoldLineWidth="2" Classes="Single" />
  <Border Classes="Blockquote" xmlns="">
    <Border.Child>
      <StackPanel Classes="Blockquote">
        <Border Classes="CodeBlock">
          <Border.Child>
            <ScrollViewer Classes="CodeBlock">
              <ScrollViewer.Content>
                <TextBlock Text="asdf&#xA; ghij" Classes="CodeBlock" />
              </ScrollViewer.Content>
            </ScrollViewer>
          </Border.Child>
        </Border>
      </StackPanel>
    </Border.Child>
  </Border>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <Border Classes="CodeBlock">
        <Border.Child>
          <ScrollViewer Classes="CodeBlock">
            <ScrollViewer.Content>
              <TextBlock Text="public void Main(){&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;}" Classes="CodeBlock" />
            </ScrollViewer.Content>
          </ScrollViewer>
        </Border.Child>
      </Border>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <m:Rule BoldLineWidth="2" Classes="Bold" />
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="▪" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="one public void Main(){" />
      </c:CTextBlock>
      <Border Classes="CodeBlock">
        <Border.Child>
          <ScrollViewer Classes="CodeBlock">
            <ScrollViewer.Content>
              <TextBlock Text="}" Classes="CodeBlock" />
            </ScrollViewer.Content>
          </ScrollViewer>
        </Border.Child>
      </Border>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <m:Rule BoldLineWidth="2" Classes="Bold" />
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="▪" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="one" />
      </c:CTextBlock>
      <Border Classes="CodeBlock">
        <Border.Child>
          <ScrollViewer Classes="CodeBlock">
            <ScrollViewer.Content>
              <TextBlock Text="public void Main(){&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;&#xA;}" Classes="CodeBlock" />
            </ScrollViewer.Content>
          </ScrollViewer>
        </Border.Child>
      </Border>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <m:Rule BoldLineWidth="2" Classes="Bold" />
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <a:TextEditor IsReadOnly="True" IsModified="True" Tag="c" Text="public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}" SelectedText="" CaretOffset="0" SelectionStart="0" SelectionLength="0" />
    </Border.Child>
  </Border>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <a:TextEditor IsReadOnly="True" IsModified="True" Tag="cs" Text="public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}" SelectedText="" CaretOffset="0" SelectionStart="0" SelectionLength="0" />
    </Border.Child>
  </Border>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <a:TextEditor IsReadOnly="True" IsModified="True" Tag="java" Text="public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}" SelectedText="" CaretOffset="0" SelectionStart="0" SelectionLength="0" />
    </Border.Child>
  </Border>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <a:TextEditor IsReadOnly="True" IsModified="True" Tag="html" Text="public void Main(){&#xA;    Console.WriteLine(&quot;`Hello World`&quot;);&#xA;}" SelectedText="" CaretOffset="0" SelectionStart="0" SelectionLength="0" />
    </Border.Child>
  </Border>
</StackPanel>
